pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -136,144 +150,163 @@ func (o *InstallOptions) prefixFor(name string) string {
 	return name[0 : len(name)-len(fmt.Sprintf("-%s", o.Revision))]
 }
 
-func (o *InstallOptions) copyContent() error {
-	// gather all secrets
+func (o *InstallOptions) copyContent(ctx context.Context) error {
+	secretPrefixes := sets.NewString(o.SecretNamePrefixes...)
+	optionalSecretPrefixes := sets.NewString(o.OptionalConfigMapNamePrefixes...)
+	configPrefixes := sets.NewString(o.ConfigMapNamePrefixes...)
+	optionalConfigPrefixes := sets.NewString(o.OptionalConfigMapNamePrefixes...)
+
+	// Gather secrets. If we get API server error, retry getting until we hit the timeout.
+	// Retrying will prevent temporary API server blips or networking issues.
 	secrets := []*corev1.Secret{}
-	for _, currPrefix := range o.SecretNamePrefixes {
-		glog.Infof("getting secrets/%s -n %s", o.nameFor(currPrefix), o.Namespace)
-		val, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(currPrefix), metav1.GetOptions{})
-		if err != nil {
-			return err
+	err := utilwait.PollImmediateUntil(200*time.Millisecond, func() (bool, error) {
+		for _, prefix := range append(secretPrefixes.List(), optionalSecretPrefixes.List()...) {
+			secret, err := o.KubeClient.CoreV1().Secrets(o.Namespace).Get(o.nameFor(prefix), metav1.GetOptions{})
+			if err != nil {
```

Comment:
are you sure we want to repeat on NotFound?

Reply:
why not? it might be that the secret is being recreated by other controller or we waiting for some process to add that secret.